modules/core/src/helpers/profile.ts

```diff
@@ -58,12 +58,17 @@ export const resolveUserDisplayName = (
  *
  * @param username - Username of the user.
  * @param userStoreDomain - User store domain of the user.
+ * @param primaryUserStoreDomainName - Domain name of the primary user store.
  * @returns resolved username
  */
-export const resolveUsername = (username: string, userStoreDomain: string): string => {
+export const resolveUsername = (
+    username: string,
+    userStoreDomain: string,
+    primaryUserStoreDomainName: string
```

Comment:
Instead of passing primaryUserStoreDomainName as a mandatory parameter, can't we take it from the redux store or some other config if possible?

PS:  There is no straight way to do this at the moment. The only way to do this, afaiu, is to move the shared usage of redux store to core module and use it here. 

WDYT? @brionmario

Reply:
Since this is a `.ts` file, I believe we can't directly access the Redux store from here. If we plan to make such an improvement, it would be better to address it as a separate effort, as there are many places in the extensions where we use the window object to retrieve configurations.

Reply:
> If we plan to make such an improvement, it would be better to address it as a separate effort, as there are many places in the extensions where we use the window object to retrieve configurations.

Yes, agreed. The comment was just to initiate a discussion :)